optimum/habana/transformers/generation/utils.py

```diff
@@ -536,11 +537,15 @@ def generate(
         if generation_config.ignore_eos is None:
             generation_config.ignore_eos = kwargs.get("ignore_eos", lazy_mode)
         num_virtual_tokens = kwargs.pop("num_virtual_tokens", 0)
+        for key in kwargs.keys():
+            if key == "max_new_tokens":
+                kwargs["max_new_tokens"]=generation_config.max_new_tokens
```

Comment:
What are these 3 lines, why do we need them?

Also this can be condensed to :
```
if "max_new_tokens" in kwargs:
    kwargs["max_new_tokens"]=generation_config.max_new_tokens
```